src/lib/DarkMode.svelte

```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+
+	let dark: boolean;
+	let hidden = true;
+	onMount(() => {
+		dark = document.documentElement.classList.contains('dark');
+		hidden = false;
+		const matcher = window.matchMedia('(prefers-color-scheme: dark)');
+		matcher.addEventListener('change', handleChange);
+		return () => matcher.removeEventListener('change', handleChange);
+	});
+	function handleChange({ matches: dark }: MediaQueryListEvent) {
+		if (!localStorage.theme) {
+			setMode(dark);
+		}
+	}
+	function toggle() {
+		setMode(!dark);
+	}
+	function setMode(value: boolean) {
+		dark = value;
+		if (dark) {
+			document.documentElement.classList.add('dark');
+		} else {
+			document.documentElement.classList.remove('dark');
+		}
+		localStorage.theme = dark ? 'dark' : 'light';
+		if (window.matchMedia(`(prefers-color-scheme: ${localStorage.theme})`).matches) {
+			localStorage.removeItem('theme');
+		}
+	}
+</script>
+
+<svelte:head>
+	<script>
+		if (
+			localStorage.theme === 'dark' ||
+			(!localStorage.theme && window.matchMedia('(prefers-color-scheme: dark)').matches)
+		) {
+			document.documentElement.classList.add('dark');
+		} else {
+			document.documentElement.classList.remove('dark');
+		}
+	</script>
+</svelte:head>
+
+<button
```

Comment:
We should create an issue to create a Switch component, then use that here. Theres a few accessibility requirements to consider for switches that this currently doesn't handle - https://www.w3.org/WAI/ARIA/apg/patterns/switch/

Unfortunately the switch in headless ui (https://captaincodeman.github.io/svelte-headlessui/switch/) is not implemented yet - might be a good contribution to make